client/js/socket-events/search.js

```diff
@@ -0,0 +1,42 @@
+"use strict";
+
+const $ = require("jquery");
+const socket = require("../socket");
+const render = require("../render");
+const messages = $("#search-query .results .messages");
+const form = $("#search-query form");
+const select = $("#search-query form select");
+
+socket.on("search-networks-channels", function(data) {
```

Comment:
I can see how this could be useful to search logs for channels that you are no longer a part of, but I also think such a niche feature warrants code this convoluted (which also runs a DISTINCT on the database).

This could be simplified a lot by simply taking networks/channels from the sidebar UI.

Reply:
So you would be able to filter only on the still opened networks, channels, queries... and If you search for something that is not opened anymore (for ex. a past chat you had in a query window) you will have to select the network entry or the "all network / all channels" entry? 

I'm also not a huge fan of the DISTINCT query (and this event).

Everybody is fine with that?